### DISCUSSION TITLE
Pre-release 3.5.0 rc1

### DISCUSSION THREAD

**Discussion by mrbean-bremen** [MeVisLab/pythonqt#180] [Announcements]
This is a pre-release for version 3.5.0, which will support Qt 5.15 and Qt 6.

The generator has been adapted to be able to produce code for these versions, and to generate stable output.
The PythonQt library should be able to work with Qt 5 and 6 if compiled using these versions.
No generated sources are added for these versions. If needed, they may be added before the release.

Please test the version for your use cases, so that any fixes can be incorporated before the release (which is planned somewhere at the beginning of 2024). You can discuss the release in this issue.

<hr /><em>This discussion was created from the release <a href='https://github.com/MeVisLab/pythonqt/releases/tag/v3.5.0-rc.1'>Pre-release 3.5.0 rc1</a>.</em>

**Comment by he-hesce** [comment]
@mrbean-bremen: I have requested time in February (which may be too late in your release schedule) to see if the release candidate integrates and works in our environment. I will then test on CentOS 7 and Rocky Linux 9. Though we may stay on the current stable release until we need to support Rocky Linux 10 (including whatever Python version it will ship) in two years. Current stable release works great so if it ain't broken, we won't upgrade unnecessarily.

**Comment by mrbean-bremen** [comment]
> which may be too late in your release schedule

If you find any problem, we can always make a patch release with the fix.

**Reply by he-hesce** [reply]
We may need to look at unofficial support in our product for Ubuntu Linux 22.04 LTS in May this year and this could possibly require a newer PythonQt depending on which Python version Ubuntu ship, so this may require us to upgrade PythonQt.

**Comment by he-hesce** [comment]
I found some time earlier than expected...

I have integrated 3.5.0-rc.1 into our in-house bespoke scientific visualization software system. I used the 5.6 pre-generated wrappers (disabling the 5.11 ones) on both the two platforms I tested on as they are the latest containing WebKit which we still use with Qt 5.9.7 on CentOS 7.9 and Qt 5.15.9 on Rocky Linux 9.3.

Qt WebKit is maintained/updated by the original developer after Qt Company abandoned in after 5.6 and still ships with latest RHEL and Ubuntu. We rely on it as WebEngine is missing features or breaks existing functionality we use with WebKit. In addition Qt WebEngine was removed from later CentOS 7 releases as it (latest Chrome needed for security reasons) no longer compiles so it is not even an option on CentOS 7 anymore.

Compilation is fine on both platforms and our Python-based integration tests pass using Python 2.7 on CentOS 7 and Python 3.9 on Rocky Linux 9.

It would be nice if pre-generated wrappers were provided for 5.15 in the future as then one can access newer Qt features from 5.15 in one's embedded Python programs. Right now we are stuck with 5.6 API which is fine with Qt 5.9 on CentOS 7 but would eventually be nice to have when we drop CentOS 7 support. Rocky Linux 8, 9, and Ubuntu 22.04 ship Qt 5.15 while Ubuntu 20.04 is shipping Qt 5.12. If the pre-generated wrappers included WebKit that would then be awesome as otherwise we could not use them.

I haven't yet looked into generating wrappers at every build. Perhaps if this was done by the PythonQt build process itself (enabled with a build option, overriding the pre-generated ones) I would try it out. I am hesitant otherwise to fiddle with this on our supported platforms given that pre-generated 5.6 wrappers are extremely stable and I do not want to introduce a potential vector of instability and build issue.

I tested our in-house bespoke scientific visualization software system (GUI-based with Python as embedded scripting language) with PythonQt 3.5.0-rc.1 on CentOS 7.9 and Rocky Linux 9.3. I did not (yet) test on Rocky Linux 8, Ubuntu 20.04, nor Ubuntu 22.04.

**Reply by usiems** [reply]
See merge request #183 where I added the Qt 5.15 bindings. Perhaps you could check it out?

**Comment by he-hesce** [comment]
@usiems : I could do that. Will you tag an rc-2 on commit?

**Comment by he-hesce** [comment]
@usiems, @mrbean-bremen:

I have integrated 3.5.0-rc.2 into our in-house bespoke scientific visualization software system. The 5.15 pre-generated wrappers were used/found on Rocky Linux 9.3. Compilation is fine and our Python-based integration tests pass using Python 3.9 on Rocky Linux 9.3.

As nothing should have changed with the 5.6 pre-generated wrappers since rc-1, I did not perform any testing on any other platform nor with any other Qt version than Rocky Linux 9.3, Qt 5.15.9, and Python 3.9.18.

**Reply by mrbean-bremen** [reply]
We haven't had problems of our own with that version using Qt 6.6 so far, we will wait a bit until a final release tag. The folks at [commontk](https://github.com/commontk/PythonQt) have also merged most of the changes into their fork, we'll also wait a bit if they find any problems.

**Comment by mrbean-bremen** [comment]
@jcfr: I saw your latest commits in commontk (related to bringing back `QMatrix` for 5.15 and warning fixes). Are you planning to merge these (or any similar fixes you may add) back upstream? From my point of view that would certainly make sense.

**Reply by jcfr** [reply]
Thanks for the note.
And yes, I will definitely upstream the changes.
